src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPoint.java

```diff
@@ -61,4 +76,105 @@ public void resetCoords(double x, double y) {
     public Point asGeometry() {
         return new Point(getX(), getY());
     }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains WKT or coordinates.
+     *
+     * @param value  input String which needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     */
+    public XYPoint resetFromString(String value, final boolean ignoreZValue) {
+        if (value.toLowerCase(Locale.ROOT).contains("point")) {
+            return resetFromWKT(value, ignoreZValue);
+        }
+        return resetFromCoordinates(value, ignoreZValue);
+    }
+
+    /**
+     * Set x and y coordinates of XYPoint if input contains coordinates.
+     *
+     * @param value  input String which contains coordinates that needs to be parsed and validated
+     * @param ignoreZValue  boolean parameter which decides if z coordinate needs to be ignored or not
+     * @return XYPoint after setting the x and y coordinates
+     */
+    public XYPoint resetFromCoordinates(String value, final boolean ignoreZValue) {
+        String[] values = value.split(",");
+        final double x;
+        final double y;
+        int numOfValues = values.length;
+
+        if (numOfValues > 3) {
+            throw new OpenSearchParseException("expected 2 or 3 coordinates, but found: [{}]", values.length);
+        }
+
+        if (numOfValues > 2) {
+            GeoPoint.assertZValue(ignoreZValue, Double.parseDouble(values[2].trim()));
```

Comment:
We should not use GeoPoint here.

Reply:
It is actually kind of util function, so instead of replicating the code I tried to use the one in geopoint
```
public static double assertZValue(boolean ignoreZValue, double zValue) {
        if (!ignoreZValue) {
            throw new OpenSearchParseException("Exception parsing coordinates: found Z value [{}] but [{}] parameter is [{}]", new Object[]{zValue, Names.IGNORE_Z_VALUE, ignoreZValue});
        } else {
            return zValue;
        }
    }

Reply:
Do you think we still need to implement it again ?

Reply:
Yeah.